packages/cli/core/src/config/createLoadedConfig.ts

```diff
@@ -20,26 +32,84 @@ export const assignPkgConfig = (
     return undefined;
   });
 
+/**
+ * A modern config can export a function or an object
+ * If it's a function, it will be called and return a config object
+ */
+async function getConfigObject(config?: ConfigParams) {
+  if (typeof config === 'function') {
+    return (await config(0)) || {};
```

Comment:
The `config(0)` is an incorrect implementation, I will make another PR to fix it.